libwallet/assets/dcr/dex_asset.go

```diff
@@ -0,0 +1,511 @@
+// This code is available on the terms of the project LICENSE.md file, and as
+// terms of the BlueOak License. See: https://blueoakcouncil.org/license/1.0.0.
+
+package dcr
+
+// Note: Most of the code here is a copy-pasta from:
+// https://github.com/decred/dcrdex/blob/master/client/asset/dcr/spv.go
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math"
+	"sort"
+	"strings"
+
+	dexasset "decred.org/dcrdex/client/asset"
+	dexdcr "decred.org/dcrdex/client/asset/dcr"
+	"decred.org/dcrdex/dex"
+	walleterrors "decred.org/dcrwallet/v3/errors"
+	walletjson "decred.org/dcrwallet/v3/rpc/jsonrpc/types"
+	dcrwallet "decred.org/dcrwallet/v3/wallet"
+	"github.com/crypto-power/cryptopower/libwallet/assets/wallet"
+	"github.com/crypto-power/cryptopower/libwallet/utils"
+	"github.com/decred/dcrd/blockchain/stake/v5"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	chainjson "github.com/decred/dcrd/rpc/jsonrpc/types/v4"
+	"github.com/decred/dcrd/txscript/v4"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
+	"github.com/decred/dcrd/wire"
+)
+
+const (
+	walletTypeSPV = "SPV"
+)
+
+// DEXAsset wraps *Asset and implements dexdcr.Wallet.
+type DEXAsset struct {
+	asset *Asset
```

Comment:
Change this to `wallet *wallet.Wallet`. Use direct methods of the upstream dcrwallet object. That should make this file look more closely like the upstream dex spv file.